spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -625,7 +626,8 @@ public final void stop(boolean wait) {
 
 	@Override
 	public void enforceRebalance() {
-		this.enforceRebalanceRequested.set(true);
+		throw new UnsupportedOperationException("enforceRebalance() is supposed to be implemented " +
+				"by concrete implementations only.");
```

Comment:
This is redundant: you say the same in the `default` method of the `MessageListenerContainer` interface.